internal/controller/component/controller_conditions.go

```diff
@@ -24,16 +24,39 @@ import (
 	"github.com/choreo-idp/choreo/internal/controller"
 )
 
-// ReasonComponentCreated is the reason used when a component is created/ready
-const ReasonComponentCreated controller.ConditionReason = "ComponentCreated"
+const (
+	// ConditionCreated represents whether the component is created
+	ConditionCreated controller.ConditionType = "Created"
+	// ConditionFinalizing represents whether the component is being finalized
+	ConditionFinalizing controller.ConditionType = "Finalizing"
+)
+
+const (
+	// ReasonComponentCreated is the reason used when a component is created/ready
+	ReasonComponentCreated controller.ConditionReason = "ComponentCreated"
+
+	// ReasonComponentDeletingDeploymentTracks is the reason used when a component's dependents are being deleted'
+	ReasonComponentDeletingDeploymentTracks controller.ConditionReason = "ComponentDeletingDeploymentTracks"
+)
 
 // NewComponentCreatedCondition creates a condition to indicate the component is created/ready
 func NewComponentCreatedCondition(generation int64) metav1.Condition {
 	return controller.NewCondition(
-		controller.TypeCreated,
+		ConditionCreated,
 		metav1.ConditionTrue,
 		ReasonComponentCreated,
 		"Component is created",
 		generation,
 	)
 }
+
+// NewComponentCleanDeploymentTracksCondition creates a condition to indicate the component is being finalized
+func NewComponentCleanDeploymentTracksCondition(generation int64) metav1.Condition {
+	return controller.NewCondition(
+		ConditionFinalizing,
+		metav1.ConditionTrue,
+		ReasonComponentDeletingDeploymentTracks,
+		"Component's DeploymentTracks are being cleaned",
```

Comment:
```suggestion
		"Component is being finalized.",
```

Reference: deployment controller deletion.